battery/battery.h

```diff
@@ -0,0 +1,44 @@
+#ifndef BATTERY_H
+#define BATTERY_H
+
+#include <QObject>
+#include <QVariant>
+#include <QDebug>
+#include <QColor>
+
+class Battery : public QObject
+{
+    Q_OBJECT
+    Q_PROPERTY(int batteryCharge READ getValue WRITE setValue)
```

Comment:
if you add NOTIFY to your Q_PROPERTY, the property (in this case batteryCharge can be "observed" by QML and it will be notified whenever a change is made

Reply:
I also suggest to use more descriptive words, setValue could be any value

```suggestion
    Q_PROPERTY(int batteryPercentage READ getBatteryPercentage WRITE setBatteryPercentage NOTIFY batteryPercentageChanged)
```